checks/store/legacyrank/legacy_rank.go

```diff
@@ -0,0 +1,99 @@
+package legacyrank
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/csv"
+	"errors"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var ErrNotFound = errors.New("domain not found")
+
+type Getter interface {
+	GetLegacyRank(domain string) (int, error)
+}
+
+type GetterFunc func(domain string) (int, error)
+
+func (f GetterFunc) GetLegacyRank(domain string) (int, error) {
+	return f(domain)
+}
+
+type InMemoryStore struct{}
+
+var once sync.Once
+var data map[string]int //map of domain to rank
+
+func NewInMemoryStore() *InMemoryStore {
+	return &InMemoryStore{}
+}
+
+func (s *InMemoryStore) GetLegacyRank(url string) (int, error) {
+	once.Do(func() {
+		var err error
+		data, err = load()
+		if err != nil {
+			log.Println(err)
+		}
+	})
+
+	rank, ok := data[url]
+	if !ok {
+		return -1, ErrNotFound
+	}
+	return rank, nil
+}
+
+func load() (map[string]int, error) {
```

Comment:
And maybe around error handling too. As it's beeing logged but looks like it's not returned to the caller.

Maybe error handling is something i can think about at a more global level, I'm not sure how that typically works with Go, I'd need to look into it. In the JS world, I'd have a reusable error class/function, which is called whenever there's an error. So that error handling is consistent and observability/monitoring can just be implemented in one place.

Reply:
Erros in golang are handled when the happen not at global level. 
Errors are always bubbled up. here errors are returned to the caller which in this case logs the error.